### PR TITLE
P3 associations

### PR DESCRIPTION
**Pull request by ahangarha**
In this PR I have implemented:
- One to Many relationship between `classroom` and `student` classes
- Many to Many relationship between `book` and `person` classes through `rental` class.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
```
Comment (elisha2kyakpo1) [inline book.rb]: - [x] A rental belongs to a book and a book has many rentals, to add a rental we should initiate a rental class, which takes three parameters, date, book, and person. So we can do something like this below:
```ruby
  def add_rental(person, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 2]
```diff
@@ -19,6 +20,10 @@ def correct_name
     @name
   end
 
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (elisha2kyakpo1) [inline person.rb]: - [x] Here we will do the same thing as in a book class. The rental belongs to a person.
```ruby
  def add_rental(book, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
    Rental.new(date, self, book)
  end
```

The self refers to the person instance

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @ahangarha,

You have done a great job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
### To highlight

- The PR title and description look professional 👍
- No linter errors 👍
- Commit messages are clear and descriptive 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Documenting commands or steps required to run the project on a user's local machine is the best way to avoid running into issues, even if that is necessary at this stage.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (ahangarha) [reply person.rb]: @elisha2kyakpo1 Thank you for your review and suggestions. I was not sure how it should be implemented. I thought we make new rental thought rental class directly which apparently was wrong. It makes more sense to do it the way you suggest. 

But what I see in [the tutorial](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md) is like what I did (which is wrong I think). Do you think the tutorial needs modification?

```rb
visit_3 = Visit.new("2017-11-11", spider, vet_john)
visit_4 = Visit.new("2017-10-10", spider, vet_maria)
```

Comment (ahangarha) [reply person.rb]: But also I think there is no contradiction between the two. We can/should have both. Right?

Comment (elisha2kyakpo1) [reply person.rb]: @ahangarha, you are right, looking at the tutorial we can say a rental is an object (what you wrote) which makes sense. I will take a closer look at the tutorial once more.

Comment (ahangarha) [reply person.rb]: I think you were right. I have added the changes and now I think it looks good

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Mostafa, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**